### PR TITLE
Fix Customer Group Type and RuleChecker

### PR DESCRIPTION
**Pull request by tchapi**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | 
| License         | MIT

Fix the CustomerGroupType and RuleChecker

### CODE REVIEW

File: src/Sylius/Component/Core/Promotion/Checker/Rule/CustomerGroupRuleChecker.php [hunk 1]
```diff
@@ -39,7 +39,11 @@ public function isEligible(PromotionSubjectInterface $subject, array $configurat
             return false;
         }
 
-        if ($configuration['groups'] === $customer->getGroup()->getId()) {
+        if (is_null($customer->getGroup())) {
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Promotion/Checker/Rule/CustomerGroupRuleChecker.php]: We have the following convention for checking for `null` value: `if (null === $customer->getGroup()) {`

File: src/Sylius/Component/Core/Promotion/Checker/Rule/CustomerGroupRuleChecker.php [hunk 2]
```diff
@@ -39,7 +39,11 @@ public function isEligible(PromotionSubjectInterface $subject, array $configurat
...
+            return false;
+        }
+
+        if ($configuration['group'] === $customer->getGroup()->getId()) {
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Promotion/Checker/Rule/CustomerGroupRuleChecker.php]: We should use `code`, same as we do in `TaxonRuleChecker` (can be done in a separate PR).

File: src/Sylius/Component/Core/Promotion/Checker/Rule/CustomerGroupRuleChecker.php (hunk 1 above)
Comment (tchapi) [inline src/Sylius/Component/Core/Promotion/Checker/Rule/CustomerGroupRuleChecker.php]: alright, changed

**Comment by pjedrzejewski** [comment]
Thanks @tchapi! Regarding using the code, we can handle that in a separate PR or when covering this functionality with scenarios.

**Comment by tchapi** [comment]
Yes, that's what I thought too because it's not really related to this PR in particular

**Pull request merged**